examples/tv-app/android/java/TVApp-JNI.cpp

```diff
@@ -183,3 +185,68 @@ JNI_METHOD(void, setChipDeviceEventProvider)(JNIEnv *, jobject, jobject provider
 {
     DeviceCallbacks::NewManager(provider);
 }
+
+#if CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
+class MyPincodeService : public PincodeService
+{
+    uint32_t FetchCommissionPincodeFromContentApp(uint16_t vendorId, uint16_t productId, CharSpan rotatingId) override
+    {
+        return ContentAppPlatform::GetInstance().GetPincodeFromContentApp(vendorId, productId, rotatingId);
+    }
+};
+MyPincodeService gMyPincodeService;
+
+class MyPostCommissioningListener : public PostCommissioningListener
+{
+    void CommissioningCompleted(uint16_t vendorId, uint16_t productId, NodeId nodeId, Messaging::ExchangeManager & exchangeMgr,
+                                SessionHandle & sessionHandle) override
+    {
+
+        ContentAppPlatform::GetInstance().ManageClientAccess(
+            exchangeMgr, sessionHandle, vendorId, GetDeviceCommissioner()->GetNodeId(), OnSuccessResponse, OnFailureResponse);
+    }
+
+    /* Callback when command results in success */
+    static void OnSuccessResponse(void * context)
+    {
+        ChipLogProgress(Controller, "OnSuccessResponse - Binding Add Successfully");
+        CommissionerDiscoveryController * cdc = GetCommissionerDiscoveryController();
+        if (cdc != nullptr)
+        {
+            cdc->PostCommissioningSucceeded();
+        }
+    }
+
+    /* Callback when command results in failure */
+    static void OnFailureResponse(void * context, CHIP_ERROR error)
+    {
+        ChipLogProgress(Controller, "OnFailureResponse - Binding Add Failed");
+        CommissionerDiscoveryController * cdc = GetCommissionerDiscoveryController();
+        if (cdc != nullptr)
+        {
+            cdc->PostCommissioningFailed(error);
+        }
+    }
+};
+
+MyPostCommissioningListener gMyPostCommissioningListener;
+
+void TvAppJNI::InitializeCommissioner(JNIMyUserPrompter * userPrompter)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE
+    chip::DeviceLayer::StackLock lock;
+    CommissionerDiscoveryController * cdc = GetCommissionerDiscoveryController();
+    if (cdc != nullptr && userPrompter != nullptr)
+    {
+        cdc->SetPincodeService(&gMyPincodeService);
+        cdc->SetUserPrompter(userPrompter);
+        cdc->SetPostCommissioningListener(&gMyPostCommissioningListener);
+    }
+
+    ChipLogProgress(AppServer, "Starting commissioner");
+    InitCommissioner(CHIP_PORT + 2 + 10, CHIP_UDC_PORT);
```

Comment:
what is + 2 + 10 ?

Could we create a named constant and maybe have some comment why this particular port was chosen?

Reply:
Created issue #22561 to track this.